docs/madr/decisions/039-transition-to-new-policies.md

```diff
@@ -0,0 +1,644 @@
+          description: |
+            An array of field names that also should be included in the response. By default, only 'xds' field is set by the API server. 
+          schema:
+            type: array
+            items:
+              type: string
+              enum: [diff]
+      responses:
+        '200':
+          description: Successfully retrieved snapshot.
+          content:
+            application/json:
+              schema:
+                type: object
+                required: [xds]
+                properties:
+                  xds:
+                    description: The raw XDS config as an inline JSON object
+                    type: object
+                  diff:
+                    description: |
+                      When '?include=diff' is specified, the field is set to an array of json patches between the current XDS config
+                      and XDS config that takes into account policies with 'kuma.io/effect: shadow'.
+                    type: array
+                    items:
+                      type: object
+                      required:
+                        - op
+                        - path
+                      properties:
+                        op:
+                          type: string
+                          description: Operation to be performed.
+                          enum: [add, remove, replace, move, copy, test]
+                        path:
+                          type: string
+                          description: A JSON Pointer path indicating the part of the document to operate on.
+                        value:
+                          type: object
+                          description: The value to be used within the operations.
+        '400':
+          description: Bad request. Parameters were invalid.
+        '404':
+          description: The specified mesh/name combination was not found.
+```
+
+Endpoint `/_rules` is getting the same `shadow` and `include` query parameters.
+
+### UX
+
+We can use a CLI tool like [jd](https://github.com/josephburnett/jd?tab=readme-ov-file#examples) to visualize the structural diff.
+The goal is to avoid dependencies on `jd` in the codebase, but to provide examples in the documentation that enable users to see diffs conveniently, i.e:
+
+```shell
+kumactl inspect dataplane dpp-1 --config-dump --shadow > with-shadow.json && jd <(kumactl inspect dataplane dpp-1 --config-dump) with-shadow.json
+```
+
+### Implementation
+
+Implementation wise, shadow mode requires generating 2 snapshots: one with and without the new policy. 
+By comparing these snapshots we can generate a list of jsonPatches that would be applied to the Envoy configuration. 
+
+Depending on how much time do we want to spend on the implementation, we can choose to implement shadow mode for Zone CP or Global CP.
+Implementing shadow mode for Global CP requires either generating snapshots on Global CPs or forwarding the request to Zone CPs.
```

Comment:
So we said this wasn't an option right? Should this be persisted in the MADR?

Reply:
Right, fixed it